lib/asyncapiSchemaFormatParser.js

```diff
@@ -81,17 +81,14 @@ function getValidator(version) {
  * 
  * @private
  * @param {Object} asyncapiSchema AsyncAPI specification JSON Schema
- * @param {Object} version AsyncAPI version.
  * @returns {Object} valid JSON Schema document describing format of AsyncAPI-valid schema for message payload
  */
-function preparePayloadSchema(asyncapiSchema, version) {
-  const payloadSchema = `http://asyncapi.com/definitions/${version}/schema.json`;
+function preparePayloadSchema(asyncapiSchema) {
   const definitions = asyncapiSchema.definitions;
-  // Remove the meta schemas because it is already present within Ajv, and it's not possible to add duplicate schemas.
-  delete definitions['http://json-schema.org/draft-07/schema'];
-  delete definitions['http://json-schema.org/draft-04/schema'];
+
+  //we return $ref and "definitions" and not only "definitions.schema" because some properties of "schema" refer to other definitions, like json schema one
```

Comment:
adding a comment for the future as I forgot why we did it like this in the first place and was like: "how the hell does it work this way, I will change to definitions.schema" 😄